### PR TITLE
Lazy load PDO object

### PR DESCRIPTION
**Pull request by camcima**
With this feature, the application will only connect to the database when it's first needed.

### CODE REVIEW

**Review by fernandocarletti — DISMISSED** [review]

**Comment by fernandocarletti** [comment]
We need to update the `.php_cs` with the newer configuration options.

**Comment by nesl247** [comment]
Updated in #18 which I am waiting to merge.

**Comment by nesl247** [comment]
If you can rebase against master, the CI process will stop failing.

File: tests/Adapter/PdoAdapterTest.php [hunk 1]
```diff
@@ -54,9 +54,24 @@ public function testIsConstructing(): void
...
+    public function testIsPdoObjectLazilyInstantiated(): void
+    {
+        $testAdapter = new PdoAdapter(DatabaseManager::DRIVER_MYSQL, $this->driverOptions, DatabaseManager::ROLE_MASTER);
+
+        $adapterPdo = Assert::readAttribute($testAdapter, 'pdo');
+        $this->assertNull($adapterPdo);
+
+        $testAdapter->execute('SELECT 1');
+
+        // @var $adapterPdo PDO
```
Comment (nesl247) [inline tests/Adapter/PdoAdapterTest.php]: This isn't a valid docblock.

File: src/Adapter/PdoAdapter.php [hunk 2]
```diff
@@ -233,6 +238,15 @@ public function escapeValue(string $value): string
         }
     }
 
+    protected function getPdo(): \PDO
```
Comment (nesl247) [inline src/Adapter/PdoAdapter.php]: We import everything.

**Review by nesl247 — CHANGES_REQUESTED** [review]

**Review by nesl247 — APPROVED** [review]

**Pull request merged**